compiler/optimizer/src/main/java/org/apache/nemo/compiler/optimizer/NemoOptimizer.java

```diff
@@ -113,6 +123,23 @@ public IRDAG optimizeAtRunTime(final IRDAG dag, final Message message) {
     return optimizationPolicy.runRunTimeOptimizations(dag, message);
   }
 
+  /**
+   * Operations to be done prior to the Compile-Time Optimizations.
+   * @param dag the DAG to process.
+   * @param policy the optimization policy to optimize the DAG with.
+   */
+  private void beforeCompileTimeOptimization(final IRDAG dag, final Policy policy) {
+    if (policy.getClass().getName().contains("XGBoost")) {
+      clientRPC.send(ControlMessage.DriverToClientMessage.newBuilder()
+        .setType(ControlMessage.DriverToClientMessageType.LaunchOptimization)
+        .setOptimizationType("xgboost")
+        .setDataCollected(ControlMessage.DataCollectMessage.newBuilder()
+          .setData(dag.irDAGSummary() + OptimizerUtils.filterEnvironmentTypeString(this.environmentType))
```

Comment:
`+ "-" + this.environmentType`? why do we need `filterEnvironmentTypeString`?

Reply:
It's been removed.